src/sortStudents.ts

```diff
@@ -1,16 +1,65 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades'
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyStudents = [...students];
+  const sortedStudents = copyStudents.sort((student, nextStudent) => {
+    switch (sortBy) {
+      case SortType.AverageGrade: {
+        const studentAverageGrade = student[sortBy]
+          .reduce((total, mark) => total + mark) / student[sortBy].length;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+        const nextStudentAverageGrade = nextStudent[sortBy]
+          .reduce((total, mark) => total + mark) / nextStudent[sortBy].length;
+
+        return order === 'asc'
+          ? studentAverageGrade - nextStudentAverageGrade
+          : nextStudentAverageGrade - studentAverageGrade;
```

Comment:
Create separate function `getAverageGrades` for find average grade and use it here.
```suggestion
        return order === 'asc'
          ? getAverageGrades(student[sortBy]) - getAverageGrades(nextStudent[sortBy])
          : getAverageGrades(nextStudent[sortBy]) - getAverageGrades(student[sortBy]);
```